### DISCUSSION TITLE
Allow to check configuration of all resources like ingresses

### DISCUSSION THREAD

**Discussion by fredgate** [aquasecurity/starboard#828] [Development]
I use starboard on a pre-production k8s cluster, where all our development teams deploy their Helm charts. So I would to check that all deployed resources respect some best practices.  
For that I use starboard with conftest and several selected policies. One thing that I want to be sure is that no resources use some deprecated k8s API. For that I use policies from [deprek8ion project](https://github.com/swade1987/deprek8ion/blob/master/policies/kubernetes-1.19.rego#L61).

K8s 1.19 deprecated some API that are removed in k8s 1.22. These [deprek8ion policies](https://github.com/swade1987/deprek8ion/blob/master/policies/kubernetes-1.19.rego#L61) check all these deprecations that concern for example ingresses. 

The problem is that the `ConfigAuditReportReconciler` controller does not support ingresses:
https://github.com/aquasecurity/starboard/blob/7ef9060633ba2143679710e336233d3b18ec6e52/pkg/operator/controller/configauditreport.go#L49-L65
Could we add more supported resources, like ingresses...

**Comment by danielpacak** [comment]
When it comes to configuration audits we definitely want to support all resource kinds!

Currently (v0.13.0) the Conftest plugin has a predefined list or resource kinds, but short term we can extend it easily by adding Ingress there.

Eventually we'd like to make the list of discovered resources configurable, but this requires thinking about RBAC config. For example, adding Ingress or RuntimeClass object must be synced with RBAC permissions granted to a service account used run a Starboard Operator pod. This is also somehow related to another ask from the community to run the Conftest plugin with a dedicated service account, which may be a bit simpler to configure overall.

**Comment by fredgate** [comment]
Very interesting. What is the roadmap? Can we participate?

**Reply by danielpacak** [reply]
For the upcoming release [v0.14] we are focusing or vulnerability scanning, especially scanning workloads that run containers from private or managed registries such as ECR where you don't really create imagePullSecrets but authorize a service account or cluster node to pull down container images.

Anyway, we can start thinking about configuration checkers in parallel. I created #836 as a starting point to write a minimal design document. Please let me know if anyone would like to pick it up.

[v0.14]: https://github.com/aquasecurity/starboard/milestone/18
